docs/howto/use_package_storage_v2.md

```diff
@@ -0,0 +1,79 @@
+We recommend to use proper versioning instead and follow these rules:
+* a package with `version < 1.0.0` is experimental
+* a package with version containing a prerelease tag (beta1, SNAPSHOT, next) is experimental
+
+## What is the goal of storage migration from v1 to v2?
+
+We identified a few issues in v1 design, we couldn't easily overcome or patch:
+1. Automatically release new Docker images of the Package Storage without missing packages due to a race condition
+   between CI jobs.
+2. Controle the Docker image size, which is constantly growing (as of today, >1GB). Packages with size >1GB must be served through the Package Registry too.
+3. Deprecate promotion between stages. It caused a lot of frustration for customers and most of them didn't follow the recommended promotion path
+   `snapshot -> staging -> production`.
+4. Enable validation for incoming packages (spec and signatures).
+5. Support package signatures. It wasn't possible to calculate the signature for unarchived package directories.
+
+## What should a package owner do to automatically publish their packages?
+
+### Existing packages
+
+Package revisions already deployed in the production Package Storage (present in the `production` branch of the [package-storage](https://github.com/elastic/package-storage))
+are automatically synced with the new storage. In this case we disable the validation as some older packages will not be able
+to pass current spec requirements.
+
+Sync between v1 and v2 will be enabled until we officially deprecate the v1 storage (no more PRs or promotions).
+
+### Next revisions
+
+Before we deprecate the v1 storage, package owners will have to adjust their releasing pipelines to submit packages
+to the new destination. Every package candidate should be submitted together with a corresponding signature, generated
+using the [Elastic signing pipeline](https://internal-ci.elastic.co/job/elastic+unified-release+master+sign-artifacts-with-gpg/).
+
+Here is the list of requirements and code modifications based on the `beats-ci`.
+
+#### Requirements
+
+1. CI job signing credentials (`sign-artifacts-with-gpg-job`) - use them to call the signing pipeline on
+   the `internal-ci` Jenkins instance. The pipeline will sign artifacts uploaded to the signing bucket and upload there their signatures.
+2. Signing bucket credentials (`internal-ci-gcs-plugin`) - use them to upload zipped packages to be signed
+   and download matching signatures.
+3. Package Storage GCP uploader credentials (`upload-package-to-package-storage`) - use them to upload a package candidate to the "queue" bucket.
+   The candidates will be picked by the publishing job and removed after processing.
+4. Package Storage uploader secret (`secret/gce/elastic-bekitzur/service-account/package-storage-uploader`) - use it to kick off
+   the publishing job to process the uploaded candidate.
```

Comment:
Could we provide a minimal pipeline as example of what would be needed in a project to publish a package?

This could help to identify if we can provide a one/two-liner helper that could be part of a jenkins library (or https://github.com/elastic/apm-pipeline-library/).

For example the functions of the package-storage jenkinsfile linked below could be parameterized and moved to a shared library.

Reply:
Good question, there are three important benefits while using a shared library:
1) Easy to test it with UTs.
2) Easy to generalise therefore others could benefit from.
3) Simplify the pipeline logic.

On the other hand, it might require backward compatibility if a new change is added, since you might not know what other consumers use it.

Reply:
Fortunately, it calls two APIs, signing and publishing, so I don't expect it to be super complex. At least for now.

Reply:
PR to define a minimal publishing pipeline: https://github.com/elastic/elastic-package/pull/729

Reply:
PR to implement the minimal publishing pipeline: https://github.com/elastic/elastic-package/pull/731